contracts/hooks/routing-custom/src/lib.rs

```diff
@@ -178,6 +181,15 @@ fn register(
     );
 
     for msg in msgs.clone() {
+        let recipient = HexBinary::from_hex(&msg.recipient)?;
```

Comment:
wdyt about converting msg.recipient to H256 like we discussed? it would allow us to reuse more hyperlane types

Reply:
we tried, but i think we should prune more packages to use them.
on hyperlane-core at v3 branch. we got this error
```
Error: execute error: failed to execute message; message index: 0: Error calling the VM: Error during static Wasm validation: Wasm contract requires unsupported import: "__wbindgen_placeholder__.__wbindgen_describe". Required imports: {"__wbindgen_externref_xform__.__wbindgen_externref_table_grow", "__wbindgen_externref_xform__.__wbindgen_externref_table_set_null", "__wbindgen_placeholder__.__wbindgen_describe", ... 16 more}. Available imports: ["env.abort", "env.db_read", "env.db_write", "env.db_remove", "env.addr_validate", "env.addr_canonicalize", "env.addr_humanize", "env.secp256k1_verify", "env.secp256k1_recover_pubkey", "env.ed25519_verify", "env.ed25519_batch_verify", "env.debug", "env.query_chain", "env.db_scan", "env.db_next"].: create wasm contract failed
```